Purchases/Networking/Operations/GetOfferingsOperation.swift

```diff
@@ -39,13 +39,19 @@ private extension GetOfferingsOperation {
             self.offeringsCallbackCache.performOnAllItemsAndRemoveFromCache(withCacheable: self) { callback in
                 callback.completion(nil, ErrorUtils.missingAppUserIDError())
             }
+            self.finish()
+
             return
         }
 
         let path = "/subscribers/\(appUserID)/offerings"
         httpClient.performGETRequest(serially: true,
                                      path: path,
                                      headers: authHeaders) { statusCode, response, error in
+            defer {
+                self.finish()
+            }
+
```

Comment:
Can this `defer` move above the `guard` so `self.finish()` doesn't need to be explicitly called twice?

Reply:
It seems like most of the changes have two calls like this in it. I _believe_ everything should work if we just have

```swift
defer {
  self.finish()
}
```

as the right thing in each method?

Reply:
Notice that this is inside the `httpClient` callback. We can't finish the operation until the HTTP request has finished, that's why we need 2 in this case (and in others). The first `finish` is to complete the operation in the case of an early return. Does that make sense?

Reply:
@NachoSoto I did a little sample with how `defer` works with callbacks in Playgrounds before commenting this and it seemed like the `defer` would still execute even after callback. But maybe different is happening here so its after to not rely on the `defer` before a callback 🤷‍♂️ 

![Screen Shot 2022-02-07 at 8 02 04 PM](https://user-images.githubusercontent.com/401294/152903761-02ab2ac2-715b-46bd-ae23-6fe7f2dd8952.png)

Reply:
That callback is executing synchronously and immediately, so of course it does 😅 but our API requests are asynchronous.
Try adding a `DispatchQueue.async` to the completion in `api`.